contracts/TokenGeyser.sol

```diff
@@ -488,19 +458,75 @@ contract TokenGeyser is IStaking, Ownable {
 
         totalLockedShares = totalLockedShares.add(mintedLockedShares);
 
-        require(
-            _lockedPool.token().transferFrom(msg.sender, address(_lockedPool), amount),
-            "TokenGeyser: transfer into locked pool failed"
-        );
+        lockedPool.token().safeTransferFrom(msg.sender, address(lockedPool), amount);
         emit TokensLocked(amount, durationSec, totalLocked());
     }
 
     /**
-     * @dev Moves distribution tokens from the locked pool to the unlocked pool, according to the
-     *      previously defined unlock schedules. Publicly callable.
-     * @return Number of newly unlocked distribution tokens.
+     * @dev Lets the owner rescue funds air-dropped to the staking pool.
+     * @param tokenToRescue Address of the token to be rescued.
+     * @param to Address to which the rescued funds are to be sent.
+     * @param amount Amount of tokens to be rescued.
+     */
+    function rescueFundsFromStakingPool(
+        address tokenToRescue,
+        address to,
+        uint256 amount
+    ) external onlyOwner {
+        stakingPool.rescueFunds(tokenToRescue, to, amount);
+    }
+
+    //-------------------------------------------------------------------------
+    // Private methods
+
+    /**
+     * @dev Updates time-dependent global storage state.
+     */
+    function _updateAccounting()
+        private
+        returns (uint256, uint256, uint256, uint256, uint256, uint256)
+    {
+        _unlockTokens();
+
+        // Global accounting
+        uint256 newStakingShareSeconds = block
+            .timestamp
+            .sub(lastAccountingTimestampSec)
+            .mul(totalStakingShares);
+        totalStakingShareSeconds = totalStakingShareSeconds.add(newStakingShareSeconds);
+        lastAccountingTimestampSec = block.timestamp;
+
+        // User Accounting
+        UserTotals storage totals = userTotals[msg.sender];
```

Comment:
nit: consider renaming `totals` to something like `user`. current naming is a little awkward later on when calculating `totalUserRewards` using `totals.stakingShareSeconds` as user's values, and `totalStakingShareSeconds` as total